Cannot be compiled on FreeBSD

Cannot be compiled on FreeBSD:

```
% sudo pkg install gmake gcc sdl2

% gmake                     
gcc *.c ../*.c -w -lSDL2 -lpthread -o btx_sdl
../layer2.c: In function 'layer2_connect2':
../layer2.c:38:24: error: storage size of 'their_addr' isn't known
   38 |     struct sockaddr_in their_addr;
      |                        ^~~~~~~~~~
../layer2.c:45:26: error: 'AF_INET' undeclared (first use in this function)
   45 |     if ((sockfd = socket(AF_INET, SOCK_STREAM, 0)) == -1) {
      |                          ^~~~~~~
../layer2.c:45:26: note: each undeclared identifier is reported only once for each function it appears in
../layer2.c:45:35: error: 'SOCK_STREAM' undeclared (first use in this function)
   45 |     if ((sockfd = socket(AF_INET, SOCK_STREAM, 0)) == -1) {
      |                                   ^~~~~~~~~~~
../layer2.c:52:27: error: dereferencing pointer to incomplete type 'struct in_addr'
   52 |     their_addr.sin_addr = *((struct in_addr *)he->h_addr);
      |                           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../layer2.c:55:64: error: invalid application of 'sizeof' to incomplete type 'struct sockaddr'
   55 |     if (connect(sockfd, (struct sockaddr *)&their_addr, sizeof(struct sockaddr)) == -1) {
      |                                                                ^~~~~~
gmake: *** [Makefile:19: all] Fehler 1

% uname -v
FreeBSD 12.1-RELEASE r354233 GENERIC

% gcc --version
gcc (FreeBSD Ports Collection) 9.3.0
```

Adding 

```
#include <sys/socket.h>
```

in `layer2.c`

reduces the errors to:

```
% gmake           
gcc *.c ../*.c -w -lSDL2 -lpthread -o btx_sdl
../layer2.c: In function 'layer2_connect2':
../layer2.c:40:24: error: storage size of 'their_addr' isn't known
   40 |     struct sockaddr_in their_addr;
      |                        ^~~~~~~~~~
../layer2.c:54:27: error: dereferencing pointer to incomplete type 'struct in_addr'
   54 |     their_addr.sin_addr = *((struct in_addr *)he->h_addr);
      |                           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
gmake: *** [Makefile:19: all] Fehler 1
```

--- a/layer2.c
+++ b/layer2.c
@@ -3,6 +3,8 @@
 #include <string.h>
 #include <netdb.h>
 #include <stdint.h>
+#include <sys/socket.h>
+#include <netinet/in.h>
 
 /*
  * The original Bildschirmtext service over modem supported

